### PR TITLE
docs: release notes template

### PR DESCRIPTION
**Pull request by mpoke**
## Description

Closes: NA

Add template for release notes and upgrading info.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct `docs:` prefix in the PR title
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary 
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct `docs:` prefix in the PR title
- [ ] Confirmed all author checklist items have been addressed 
- [ ] Confirmed that this PR only changes documentation
- [ ] Reviewed content for consistency
- [ ] Reviewed content for spelling and grammar
- [ ] Tested instructions (if applicable)
- [ ] Checked that the documentation website can be built and deployed successfully (run `make build-docs`)

### CODE REVIEW

File: UPGRADING.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+## [v3.0.x](https://github.com/cosmos/interchain-security/releases/tag/v3.0.0-rc0)
+
+### Upgrading to Cosmos SDK 0.47
+
+The following should be considered as complementary to [Cosmos SDK v0.47 UPGRADING.md](https://github.com/cosmos/cosmos-sdk/blob/v0.47.0-rc2/UPGRADING.md).
+
+#### Protobuf 
+
+Protobuf code generation, linting and formatting have been updated to leverage the `ghcr.io/cosmos/proto-builder:0.11.5` docker container. Replicated Security protobuf definitions are now packaged and published to [buf.build/cosmos/replicated-security](https://buf.build/cosmos/replicated-security) via CI workflows. The `third_party/proto` directory has been removed in favour of dependency management using [buf.build](https://docs.buf.build/introduction).
```
Comment (mpoke) [inline UPGRADING.md:13]: We should do this: 
> Replicated Security protobuf definitions are now packaged and published to [buf.build/cosmos/replicated-security](https://buf.build/cosmos/replicated-security) via CI workflows.

**Review by MSalopek — APPROVED** [review]

File: .github/workflows/proto-registry.yml
Comment (mpoke) [inline .github/workflows/proto-registry.yml:1]: This should do the trick.

File: UPGRADING.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# Upgrading Replicated Security
+
+This guide provides instructions for upgrading to specific versions of Replicated Security.
+
+## [v3.0.x](https://github.com/cosmos/interchain-security/releases/tag/v3.0.0-rc0)
```
Comment (smarshall-spitzbart) [inline UPGRADING.md:5]: Shouldn't this link to https://github.com/cosmos/interchain-security/tree/release/v3.0.x?

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<!--
```
Comment (smarshall-spitzbart) [inline RELEASE_NOTES.md:1]: Is the idea that we'd copy/paste this into a release description, or does GH handle the template automatically?

File: UPGRADING.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+```diff
+import (
+    // ...
+-   "github.com/gogo/protobuf/proto"
++   "github.com/cosmos/gogoproto/proto"
+    // ...
+)
+```
+
+## [v2.0.x](https://github.com/cosmos/interchain-security/releases/tag/v2.0.0)
```
Comment (smarshall-spitzbart) [inline UPGRADING.md:65]: If we add this here, we should prob remove the duplicate text that's commited to changelog

**Review by smarshall-spitzbart — COMMENTED** [review]
I understand that everyone should be pro-documentation. But we should be careful not to introduce extraneous documentation and processes that slow people down, or just make things hard to find. Staying lean is still important and there's a balance to be struck.

For example I think the info in `UPGRADING.md` could be included in existing docs like the changelog or release notes.

**Comment by mpoke** [comment]
> I understand that everyone should be pro-documentation. But we should be careful not to introduce extraneous documentation and processes that slow people down, or just make things hard to find. Staying lean is still important and there's a balance to be struck.
> 
> For example I think the info in `UPGRADING.md` could be included in existing docs like the changelog or release notes.

Thanks for the feedback @smarshall-spitzbart. The reason for this PR is to standardize our release process. Since the [SDK](https://github.com/cosmos/cosmos-sdk/blob/main/UPGRADING.md) repo is using this approach, I think it makes sense for the ICS repo to also adopt it. Note that the Gaia repo will also use this starting with v11. 

Benefits of this approach:

- More concise release notes with clear links for different stakeholders. See https://github.com/cosmos/interchain-security/releases/tag/v3.0.0-rc0 vs. https://github.com/cosmos/interchain-security/releases/tag/v2.0.0.
- Consistent structure for the ICS releases. Currently, it's difficult to find the needed info to use ICS (is it in the release notes, the changelog, or in the documentation).

File: RELEASE_NOTES.md (hunk 3 above)
Comment (mpoke) [reply RELEASE_NOTES.md:1]: Once we add goreleaser to this repo, this file will act as input. See the release process on Gaia for more details https://github.com/cosmos/gaia/blob/main/RELEASE_PROCESS.md#release-notes

File: UPGRADING.md (hunk 2 above)
Comment (smarshall-spitzbart) [reply UPGRADING.md:5]: Maybe change the text from `v3.0.x` to `v3.0.0` then, but not a biggie

**Review by smarshall-spitzbart — APPROVED** [review]

**Pull request merged**